source/ShipyardPanel.cpp

```diff
@@ -307,6 +307,24 @@ void ShipyardPanel::Sell(bool toStorage)
 	static const int MAX_LIST = 20;
 
 	int count = playerShips.size();
+	if(count == 1 && player.UuidLocked(playerShip->UUID()))
+	{
+		string message("Could not sell the ");
+		message += playerShip->Name();
+		message += ", as it is marked as locked.\n(Alt+click the ship's icon to unlock the ship.)";
+		GetUI()->Push(new Dialog(message));
+		return;
+	}
+	else if(all_of(playerShips.begin(), playerShips.end(), [this](Ship *ship)
```

Comment:
I'd just exclude all locked ships and if the final ship count is 0, the dialog would be displayed. Because currently, the confirmation message gives you the value of all ships selected to be sold, not only those that will be actually sold. Which may be misleading if you attempt to sell a valuable locked ship along with some unlocked ships.

Reply:
I would do something like:
```c++
	int selectedCount = playerShips.size();
	vector<shared_ptr<Ship>> toSell;
	for(const auto &it : playerShips)
		if(!player.UuidLocked(ship->UUID()))
			toSell.push_back(it->shared_from_this());
	if(toSell.empty())
	{
		string message("Could not sell the ");
		if(selectedCount == 1)
			message += playerShip->Name() + ", as it is";
		else
			message += "selected ships, as they are all";
		message += " marked as locked.\n(Alt+click a ship's icon to unlock the ship.)";
		GetUI()->Push(new Dialog(message));
		return;
	}

	int count = toSell.size();
	int initialCount = count;
	int maxList = (count == selectedCount) ? MAX_LIST : MAX_LIST - 3;
	string message;
	if(!toStorage)
		message = "Sell the ";
	else if(count == 1)
		message = "Sell the hull of the ";
	else
		message = "Sell the hulls of the ";
	if(count == 1)
		message += playerShip->Name();
	else if(count <= maxList)
	{
		auto it = toSell.begin();
		message += (*it++)->Name();
		--count;
		if(count == 1)
			message += " and ";
		else
		{
			while(count-- > 1)
				message += ",\n" + (*it++)->Name();
			message += ",\nand ";
		}
		message += (*it)->Name();
	}
	else
	{
		auto it = playerShips.begin();
		message += (*it++)->Name() + ",\n";
		for(int i = 1; i < maxList- 1; ++i)
			message += (*it++)->Name() + ",\n";
		message += "and " + to_string(count - (maxList- 1)) + " other ships";
	}
	int64_t total = player.FleetDepreciation().Value(toSell, day, toStorage);
	message += ((initialCount > 2) ? "\nfor " : " for ") + Format::CreditString(total) + "?";
	int locked = selectedCount - initialCount;
	if(locked)
	{
		if(locked == 1)
			message += "\nSome ships are";
		else
			message += "\nOne ship is";
		message += " marked as locked and will not be sol.\nAlt+click a ship's icon to unlock the ship.";
	}
	if(toStorage)
	{
		message += " Any outfits will be placed in storage.";
		GetUI()->Push(new Dialog(this, &ShipyardPanel::SellShipChassis, message, Truncate::MIDDLE));
	}
	else
		GetUI()->Push(new Dialog(this, &ShipyardPanel::SellShipAndOutfits, message, Truncate::MIDDLE));
```
This may require some fine turning.